addons/website/static/src/snippets/s_instagram_page/instagram_page.js

```diff
@@ -0,0 +1,89 @@
+import { Interaction } from "@website/core/interaction";
+import { registry } from "@web/core/registry";
+
+import { _t } from "@web/core/l10n/translation";
+
+// Note that Instagram can automatically detect the language of the user and
+// translate the embed.
+
+export class InstagramPage extends Interaction {
+    static selector = ".s_instagram_page";
+
+    setup() {
+        this.renderIframe();
+    }
+
+    start() {
+        // We have to setup the message listener before setting the src, because
+        // the iframe can send a message before this JS is fully loaded.
+        this.addListener(window, "message", this.onMessage);
+        const src = `https://www.instagram.com/${this.el.dataset.instagramPage}/embed`;
+        this.services.website_cookies.manageIframeSrc(iframeEl, src);
+    }
+
+    renderIframe() {
+        const iframeEl = document.createElement("iframe");
+        iframeEl.setAttribute("scrolling", "no");
+        iframeEl.setAttribute("aria-label", _t("Instagram"));
+        iframeEl.classList.add("w-100");
+
+        // In the meantime Instagram doesn't send us a message with the height,
+        // we use a formula to estimate the height of the iframe (the formula
+        // has been found with a linear regression).
+        const iframeWidth = parseInt(getComputedStyle(iframeEl).width);
+        // The profile picture is smaller when width < 432px.
+        iframeEl.height = 0.659 * iframeWidth + (iframeWidth < 432 ? 156 : 203);
+
+        this.el.querySelector(".o_instagram_container").appendChild(iframeEl);
+
+        // TODO : ...observerUnactive() / ...observerActive()
+        this.registerCleanup(() => { iframeEl.remove(); });
+    }
```

Comment:
Wht not use `this.insert(...)` ?

Reply:
We can do it depending on how the `observerUnactive` and `observerActive` are converted during the conversion. I will put a comment for this.